File: m/rls/rlb/chcvt.c
/* file -  chcvt - character conversions */
#include "m:\rid\rider.h"
#include "m:\rid\chdef.h"
#include "m:\rid\ctdef.h"
/* code -  ch_upr - convert to uppercase */
int ch_upr(
int cha )
{ if (( ct_low (cha))
  &&(!(cha == '_' || cha == '$'))) {
    cha += (_A - _a); }
  return ( cha);
} 
/* code -  ch_low - convert to lowercase */
int ch_low(
int cha )
{ if (( ct_upr (cha))
  &&(!(cha == '_' || cha == '$'))) {
    cha += (_a - _A); }
  return ( cha);
} 

File: m/rls/chcvt.r
file	chcvt - eight-bit character conversion
include rid:rider
include rid:chdef
include rid:ctdef

	chLOW := -1
	chSWI := 0
	chUPR := 1

  func	ch_cvt
	cha : int~
	cvt : int~
  is	cha &= 255
	if cha lt 128
	   if ct_upr (cha)
	      cha = ch_low (cha) if cvt ne chUPR
	   elif ct_low (cha)
	   .. cha = ch_upr (cha) if cvt ne chLOW
	else
	   if (cha ge 192) && (cha le 222) && (cha ne 215)
	      cha |= 32 if cvt ne chUPR
	   elif (cha ge 224) && (cha le 254) && (cha ne 247)
	   .. cha &= ~(32) if cvt ne chLOW
	end
	reply cha
  end
